[16.0][ADD] purchase_sale_lot_intercompany

Kev-Roche opened the pull request:
Migration of the PR:  https://github.com/OCA/multi-company/pull/481

legalsylvain commented:
Hi @Kev-Roche. Thanks for porting this module. could you take a look on red CI ? 

thanks !

metaminux commented:
Hi @sbidoul 
Tests were canceled on this PR, can we launch it again ?

sbidoul commented:
@metaminux please rebase and `git push --force-with-lease`

metaminux commented:
@Kev-Roche Can you please rebase and force push as @sbidoul stated ?

metaminux commented on purchase_sale_lot_intercompany/tests/test_purchase_sale_lot_intercompany.py:
```diff
@@ -0,0 +1,31 @@
+# @author Kévin Roche <kevin.roche@akretion.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo.addons.purchase_sale_stock_inter_company.tests.test_inter_company_purchase_sale_stock import (
+    TestPurchaseSaleStockInterCompany,
+)
```
Hi @Kev-Roche 

To avoid the conflict between black, isort and flake8 on the line length for python import, I think the least worst solution is :
```suggestion
from odoo.addons.purchase_sale_stock_inter_company.tests.test_inter_company_purchase_sale_stock import (  # noqa: B950
    TestPurchaseSaleStockInterCompany,
)
```
This should make pre-commit test green again...

Kev-Roche committed:
[16.0][ADD] purchase_sale_lot_intercompany

[ADD] prepare values

[FIX] purchase_sale_lot_intercompany : depend on "sale_order_lot_selection"

metaminux commented:
Hello @sbidoul and @legalsylvain 

This PR has been rebased and CI is full green. Can we go ahead ?

Regards,

metaminux commented:
Hello @sbidoul or @legalsylvain or any other @OCA/intercompany-maintainers 

This PR has been rebased (again), fixed and squashed and CI is full green. Can we go ahead ?

Regards,
